src/analytics.js

```diff
@@ -435,13 +436,18 @@ window.addEventListener("beforeunload", async () => {
 	await Analytics.flushEvents();
 });
 
-// Expose userland API.
-window.Altis.Analytics.updateEndpoint = Analytics.updateEndpoint;
-window.Altis.Analytics.record = (type, data = {}) =>
-	Analytics.record(
-		type,
-		{
-			attributes: getAttributes(data.attributes || {}),
-			metrics: getMetrics(data.metrics || {})
-		}
-	);
+// Expose userland API. Return empty functions if the Noop field is true.
+if (Noop) {
```

Comment:
I think this will create the same problems as before because we're only exposing the functions on the `window` object when this is false - that means something else could still try and call `window.Altis.Analytics.record()` but the function won't exist.

You would have seen this stop the default built in events from firing because it's updating the functions that they would be using but the last part of the testing should ensure that you can call `Altis.Analytics.record( 'event' )` from the console and nothing happening - but also no error thrown.

You could remove the `} else {` part so that the window object always gets updated, either with the "live" version of the functions or "no-op" version.